### PR TITLE
Add recipe for open-color

### PR DESCRIPTION
**Pull request by a13**
### Brief summary of what the package does

The package is an Emacs lisp port of the popular https://github.com/yeun/open-color color scheme. I use it for writing color schemes, so probably it will be useful for other theme makers as well.

### Direct link to the package repository

https://github.com/a13/open-color.el

### Your association with the package

Author/Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

### CODE REVIEW

File: recipes/open-color [hunk 1]
```diff
@@ -0,0 +1 @@
+(open-color :repo "a13/open-color.el" :fetcher github :files "open-color.el")
```
Comment (riscy) [inline recipes/open-color:1]: You have a couple of options here.  This is what you intended, but is corrected (`:files` must be a list):
```elisp
(open-color :repo "a13/open-color.el" :fetcher github :files ("open-color.el"))
```
but ultimately this is equivalent to the default recipe, which MELPA prefers:
```suggestion
(open-color :repo "a13/open-color.el" :fetcher github)
```

**Review by riscy — COMMENTED** [review]
Quick first pass --

`open-color.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- The `;;; Commentary` for this file is much wider than 80 characters
```

Package and license:
- Prefer equivalent recipe: `(open-color :repo "a13/open-color.el" :fetcher github)`
- unknown license -- open-color.el: Open Color color palette
- open-color.el needs *formal* license boilerplate and/or an [SPDX-License-Identifier](https://spdx.dev/ids/)

**Comment by a13** [comment]
> `unknown` license -- open-color.el: Open Color color palette

It looks like there's something wrong with the license checker

**Comment by riscy** [comment]
Thank you -- re: the license,

```
;;; License:

;; MIT License, see accompanying LICENSE file: https://github.com/a13/open-color.el/blob/master/LICENSE
```

is informal and not machine detectable.  Putting `;; SPDX-License-Identifier: MIT` in the source file (per the link I gave you) will alleviate this - or put the full MIT license boilerplate as an alternative.

Your commentary looks like it's still wrapped at ~102 characters.  This is not a huge deal, but wrapping at 80 is going to make things less painful for the lowest common denominator looking at your source file.

All that said, I took a look at what this is based on - https://github.com/yeun/open-color - and that is a project with that dev seems to already be striving to support multiple instances of open-color (e.g. there's even a .json file).  Would it make sense to try to PR this file to that greater repository?  The author says the colors are "subject to change", and this way you'd be part of whatever checks they have in place to ensure consistency across the colors?

**Comment by a13** [comment]
addressed

>and that is a project with that dev seems to already be striving to support multiple instances of open-color (e.g. there's even a .json file). 

which I use for generation

> Would it make sense to try to PR this file to that greater repository? The author says the colors are "subject to change", and this way you'd be part of whatever checks they have in place to ensure consistency across the colors?

see the motivation here https://github.com/yeun/open-color/pull/135 - I don't have time to learn some new template engine just for this project. The generator.clj script does the job just fine.

**Comment by a13** [comment]
@riscy Just curious: why do you check all the stuff manually, isn't it possible to put melpazoid into CI? I use `elisp-check` action for my projects and it works pretty good in workflows.

**Comment by riscy** [comment]
Let me get this one answered in its own reply because I think it's a good question -

> @riscy Just curious: why do you check all the stuff manually, isn't it possible to put melpazoid into CI?

I had a branch that did this but ultimately didn't merge it into the MELPA mainline (although I agree generally that the melpa/melpa CI could do a little bit more).  The details aren't that exciting, but here's why I've decided so far not to:

- it would be a running system that I'd be on the hook to maintain (and the issues with it would pile up 7 days a week)
- it doesn't work for every PR and persistent false positives in the CI would create noise and people would lose faith in its usefulness.  For example I think the popular mu4e still has an idiosyncratic install
- the code checks are by far the easiest part of the whole process. The part that is hard is (a) understanding what a package does, (b) learning/remembering if something similar to this already exists, (c) verifying whether the implementation is good and/or idiomatic, (d) negotiating exceptions with contributors
- Selfishly, I believe the review workflow where I run melpazoid locally is just faster for *me*.  Running melpazoid in CI would require me clicking around in the browser to see the results of the GitHub actions, waiting for page loads, etc.  By contrast, my current workflow is just "copy the URL of the PR I'm on" and then melpazoid reads my clipboard and runs the checks.  This also means I can tweak a few things about the recipe or the code on my machine and give better feedback

**Comment by riscy** [comment]
Thanks for the link to https://github.com/yeun/open-color/pull/135 -- that makes sense to me.  I took a look at the templates for that project too, and they seem complicated enough that I sympathize with your approach.

**Pull request merged**